overlord/devicestate/handlers_install.go

```diff
@@ -1252,45 +1298,30 @@ func (m *DeviceManager) doInstallFinish(t *state.Task, _ *tomb.Tomb) error {
 	if err := t.Get("on-volumes", &onVolumes); err != nil {
 		return err
 	}
-	logger.Debugf("starting install-finish for %q systemd on %v", systemLabel, onVolumes)
 
-	essentialTypes := []snap.Type{snap.TypeKernel, snap.TypeBase, snap.TypeGadget}
+	var encryptSetupData *install.EncryptionSetupData
+	cached := st.Cached(encryptionSetupDataKey{systemLabel})
+	if cached != nil {
+		var ok bool
+		encryptSetupData, ok = cached.(*install.EncryptionSetupData)
+		if !ok {
+			return fmt.Errorf("internal error: wrong data type under encryptionSetupDataKey")
+		}
```

Comment:
it's not an immediate concern but we need a TODO about some checking that the things we stored in encryptSetupData and the on-volumes we get here are not out of sync

Reply:
TODO added